### PR TITLE
LUFA USB descriptor cleanup

### PR DESCRIPTION
**Pull request by fauxpark**
## Description
Tidying up the LUFA descriptor code.

 - Formatted to 2 space indentation and OTBS, formatting in general
 - Made the giant configuration descriptor look nicer by aligning the RHS
 - Indented C preprocessor directives separately from C code
 - Removed a redefinition of `VERSION_BCD` that only applies to older versions of LUFA (QMK's copy is currently based on the latest stable 170418)
 - Tweaked the "out of endpoints" error message to ask nicely :)
 - Annotated the HID report descriptors to hopefully make it easier to understand how they're structured

## Types of changes

- [x] Core
- [ ] Bugfix
- [ ] New Feature
- [x] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by this PR

None.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/protocol/usb_descriptor.c [hunk 1]
```diff
@@ -40,957 +40,950 @@
...
-
-            .TotalConfigurationSize = sizeof(USB_Descriptor_Configuration_t),
-            .TotalInterfaces        = TOTAL_INTERFACES,
-
-            .ConfigurationNumber    = 1,
-            .ConfigurationStrIndex  = NO_DESCRIPTOR,
-
-            .ConfigAttributes       = (USB_CONFIG_ATTR_RESERVED | USB_CONFIG_ATTR_REMOTEWAKEUP),
+#ifndef USB_MAX_POWER_CONSUMPTION
+  #define USB_MAX_POWER_CONSUMPTION 500
```
Comment (skullydazed) [inline tmk_core/protocol/usb_descriptor.c]: Is there a reason behind moving this to the middle of the file?

Comment (fauxpark) [reply tmk_core/protocol/usb_descriptor.c]: I moved it here to be closer to the actual place where it's used. See below where it's referenced in the MaxPowerConsumption member of the configuration descriptor.

**Comment by fauxpark** [comment]
@skullydazed I agree it's quite intimidating if you're just looking at the final diff, but that's why I split this up into multiple commits, so it's easier to follow along. You'll find that 90% of this is just whitespace changes. (and in fact the answer to your above question can be found in the commits tab 😉)

**Comment by fauxpark** [comment]
Also, RE your comments on stream: there's no behaviour changes here. The reordering of the HID descriptors has a purpose - it's to give them a standardish structure that can be replicated for the other platforms eg. V-USB, ATSAM etc. which all have differences in their descriptors (see #4597). Currently the structure of these things seem a little like black magic so I'm attempting to clarify how they work a little.

**Comment by skullydazed** [comment]
I have changed the base from `master` to `future_unformatted` as part of the [Aug 2019 Breaking Change](https://github.com/qmk/qmk_firmware/issues/6146).

**Pull request merged**